src/datasets/load.py

```diff
@@ -1622,29 +1650,48 @@ def dataset_module_factory(
                     # This fails when the dataset has multiple configs and a default config and
                     # the user didn't specify a configuration name (_require_default_config_name=True).
                     try:
-                        return HubDatasetModuleFactoryWithParquetExport(
-                            path, download_config=download_config, revision=dataset_info.sha
+                        out = HubDatasetModuleFactoryWithParquetExport(
+                            path, download_config=download_config, commit_hash=commit_hash
                         ).get_module()
+                        logger.info("Loading the dataset from the Parquet export on Hugging Face.")
+                        return out
                     except _dataset_viewer.DatasetViewerError:
                         pass
                 # Otherwise we must use the dataset script if the user trusts it
                 return HubDatasetModuleFactoryWithScript(
                     path,
-                    revision=revision,
+                    commit_hash=commit_hash,
                     download_config=download_config,
                     download_mode=download_mode,
                     dynamic_modules_path=dynamic_modules_path,
                     trust_remote_code=trust_remote_code,
                 ).get_module()
-            else:
+            except EntryNotFoundError:
+                # Use the infos from the parquet export except in some cases:
+                if data_dir or data_files or (revision and revision != "main"):
+                    use_exported_dataset_infos = False
+                else:
+                    use_exported_dataset_infos = True
```

Comment:
(I also moved this from HubDatasetModuleFactoryWithoutScript directly here since I don't pass the requested`revision` anymore but the fixed `commit_hash`)